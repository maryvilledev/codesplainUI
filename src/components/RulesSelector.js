import React, { PropTypes } from 'react';
import { List, ListItem } from 'material-ui/List';
import Subheader from 'material-ui/Subheader';
import Checkbox from 'material-ui/Checkbox';

const makeListItems = (filters, onRuleSelected) => {
  const listItems = Object.keys(filters).map((filterName, index) => {
    const filter = filters[filterName]
    return (
      <ListItem
        key={filterName}
        leftCheckbox={
          <Checkbox
            checked={filter.selected}
            onCheck={() => onRuleSelected(filterName)}
          />
        }
        primaryText={filter.prettyTokenName}
      />
    )
  });
  return listItems
}

const RulesSelector = ({filters, onRuleSelected}) => {
  const listItems = makeListItems(filters, onRuleSelected)
  return (
    <List>
      <Subheader>
        Select a token type to highlight all occurences
      </Subheader>
      {listItems}
    </List>
  )
}

export default RulesSelector
