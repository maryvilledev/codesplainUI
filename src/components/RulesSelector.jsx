import React, { PropTypes } from 'react';
import Checkbox from 'material-ui/Checkbox';
import {
  List,
  ListItem
} from 'material-ui/List';
import Subheader from 'material-ui/Subheader';

const style = {
  list: {
    height: '20px',
    fontSize: '13px',
    paddingLeft: '30%',
  },
  subheader: {
    lineHeight: '',
    paddingLeft: '',
  },
};

const makeListItems = (filters, onRuleSelected) => {
  const listItems = Object.keys(filters).map((filterName) => {
    const filter = filters[filterName];
    return (
      <ListItem
        style={style.list}
        key={filterName}
        leftCheckbox={
          <Checkbox
            checked={filter.selected}
            onCheck={() => onRuleSelected(filterName)}
          />
        }
        primaryText={`${filter.prettyTokenName} (${filter.count})`}
      />
    );
  });
  return listItems;
};

const RulesSelector = ({ filters, onRuleSelected }) => {
  const listItems = makeListItems(filters, onRuleSelected);
  return (
    <List>
      <Subheader style={style.subheader}>
        Select a token type to highlight all occurences
      </Subheader>
      {listItems}
    </List>
  );
};

RulesSelector.propTypes = {
  filters: PropTypes.object.isRequired,
  onRuleSelected: PropTypes.func.isRequired,
};

export default RulesSelector;
