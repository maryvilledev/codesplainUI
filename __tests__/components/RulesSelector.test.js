import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import RulesSelector from '../../src/components/RulesSelector';

jest.mock('material-ui/Checkbox')

const mockFilters = {
  'for_stmt': {
    prettyTokenName: 'For Loops',
    count: 1,
    selected: false
  },
  'atom': {
    prettyTokenName: 'Atoms',
    count: 21,
    selected: true
  }
}

describe('<RulesSelector />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <RulesSelector
        filters={mockFilters}
        onRuleSelected={jest.fn()}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('prop: filters', () => {
    it('renders ListItems for each filter', () => {
      const wrapper = shallowWithContext(
        <RulesSelector
          filters={mockFilters}
          onRuleSelected={jest.fn()}
        />
      );
      const listItems = wrapper.find('ListItem');
      expect(listItems.length).toEqual(Object.keys(mockFilters).length);
    });
  });
});
