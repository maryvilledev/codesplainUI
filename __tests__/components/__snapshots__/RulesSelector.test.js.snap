exports[`<RulesSelector /> matches snapshot 1`] = `
<List>
  <Subheader
    inset={false}
    style={
      Object {
        "lineHeight": "",
        "paddingLeft": "",
      }
    }>
    Select a token type to highlight all occurences
  </Subheader>
  <ListItem
    autoGenerateNestedIndicator={true}
    disableKeyboardFocus={false}
    disabled={false}
    initiallyOpen={false}
    insetChildren={false}
    leftCheckbox={
      <Checkbox
        checked={false}
        disabled={false}
        labelPosition="right"
        onCheck={[Function]} />
    }
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    primaryText="For Loops (1)"
    primaryTogglesNestedList={false}
    secondaryTextLines={1}
    style={
      Object {
        "fontSize": "13px",
        "height": "20px",
        "paddingLeft": "30%",
      }
    } />
  <ListItem
    autoGenerateNestedIndicator={true}
    disableKeyboardFocus={false}
    disabled={false}
    initiallyOpen={false}
    insetChildren={false}
    leftCheckbox={
      <Checkbox
        checked={true}
        disabled={false}
        labelPosition="right"
        onCheck={[Function]} />
    }
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    primaryText="Atoms (21)"
    primaryTogglesNestedList={false}
    secondaryTextLines={1}
    style={
      Object {
        "fontSize": "13px",
        "height": "20px",
        "paddingLeft": "30%",
      }
    } />
</List>
`;
